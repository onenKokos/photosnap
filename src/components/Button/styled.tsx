import styled from 'styled-components';
import theme from '@Common/theme';

export const StyledButton = styled.button<{ theme?: 'light' | 'dark' }>`
  height: 3rem;
  padding: 0;
  width: 100%;
  grid-column-start: 1;
  grid-column-end: 4;
  display: flex;
  justify-content: center;
  align-items: center;
  text-transform: uppercase;
  border: none;
  font-weight: 600;

  background-color: ${(props) =>
    props.theme && props.theme === 'light'
      ? `${theme.colors.primaryLight}`
      : `${theme.colors.primaryDark}`};
  color: ${(props) =>
    props.theme && props.theme === 'dark'
      ? `${theme.colors.primaryLight}`
      : `${theme.colors.primaryDark}`};
`;
