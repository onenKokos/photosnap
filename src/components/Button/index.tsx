import React from 'react';
import { StyledButton } from './styled';
export interface ButtonProps {
  text: string;
  theme?: 'light' | 'dark';
  action: () => void;
}

const Button = ({ text, action, theme }: ButtonProps) => (
  <StyledButton theme={theme} onClick={action} data-testid="button">
    {text}
  </StyledButton>
);

export default Button;
