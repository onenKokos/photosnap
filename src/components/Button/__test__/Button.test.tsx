import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button, { ButtonProps } from '../';

const renderButton = (props: Partial<ButtonProps> = {}) => {
  const defaultProps: ButtonProps = {
    theme: 'dark',
    text: 'test',
    action() {
      return;
    },
  };
  return render(<Button {...defaultProps} {...props} />);
};

describe('[Component]: Button', () => {
  it('Should render correctly', () => {
    const renderer = renderButton();
    expect(renderer.container).toMatchSnapshot();
  });

  test('Should be clickable', async () => {
    const onClick = jest.fn();
    const { findByTestId } = renderButton({
      action: onClick,
    });

    const btn = await findByTestId('button');

    fireEvent.click(btn);

    expect(onClick).toHaveBeenCalled();
  });
});
