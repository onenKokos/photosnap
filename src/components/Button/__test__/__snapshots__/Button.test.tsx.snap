// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Button Should render correctly 1`] = `
<div>
  <button
    class="sc-bdfBwQ jIiGPx"
    data-testid="button"
  >
    test
  </button>
</div>
`;
